core/src/main/scala/org/apache/spark/rdd/RDD.scala

```diff
@@ -140,7 +140,7 @@ abstract class RDD[T: ClassTag](
   val id: Int = sc.newRddId()
 
   /** A friendly name for this RDD */
-  @transient var name: String = null
+  @transient var name: String = Utils.getFormattedClassName(this)
```

Comment:
Actually come to think of it, should we replace `getCreationSite` in that function with just getting the innermost scope?